apps/web/src/state/farmsV4/state/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { Token } from '@pancakeswap/swap-sdk-core'
+import { paths } from 'state/info/api/schema'
+import { safeGetAddress } from 'utils'
+import { PoolInfo } from './type'
+
+export const parseFarmPools = (
+  data:
+    | paths['/cached/pools/farming']['get']['responses']['200']['content']['application/json']
+    | paths['/cached/pools/list']['get']['responses']['200']['content']['application/json'],
+): PoolInfo[] => {
+  return data.map((pool) => {
+    return {
+      chainId: pool.chainId,
+      lpAddress: safeGetAddress(pool.id)!,
+      protocol: pool.protocol,
+      token0: new Token(
```

Comment:
can Token assign to Currency type?

Reply:
see https://github.com/pancakeswap/pancake-frontend/blob/254e13b7d24319c78a21788990fe97381fc15970/packages/swap-sdk-core/src/currency.ts#L4